src/nvim/regexp.c

```diff
@@ -3293,7 +3293,8 @@ bt_regexec_nl (
   ireg_ic = rmp->rm_ic;
   ireg_icombine = FALSE;
   ireg_maxcol = 0;
-  return bt_regexec_both(line, col, NULL) != 0;
+
+  return bt_regexec_both(line, col, NULL);
```

Comment:
I need some advice here. `bt_regexec_both` returns `long`, but this function returns `int`.

What is the proper way of handling this? Simply cast the result to `int`, or perhaps add an `assert` with a value range?